bin/angstrom/src/cli/mod.rs

```diff
@@ -248,10 +248,22 @@ pub async fn initialize_strom_components<Node: FullNodeComponents, AddOns: NodeA
 
     // I am sure there is a prettier way of doing this
     let provider = ProviderBuilder::<_, _, Ethereum>::default()
+        .with_recommended_fillers()
+        .wallet(EthereumWallet::from(
+            LocalSigner::<SigningKey>::from_bytes(&secret_key.secret_bytes().into()).unwrap()
+        ))
         .on_builtin(node.rpc_server_handles.rpc.http_url().unwrap().as_str())
         .await
         .unwrap();
```

Comment:
this is bad because reth uses a threadpool to handle calls so you could tech ddos the node by just spamming it with default reth calls

Reply:
What I need here is something that implements the alloy `Provider` trait. Unfortunately, I couldn't find anything reth native that implements the `Provider`. Thus instead I'm building the alloy native FillProvider from an https endpoint. I could use ipc too. 

DDos should not be a concern, since this is used only for `eth_sendTransaction` once per block when the bundle is ready. 

This allows for a lot less code. 

Alternatively, I could write an adapater for `AnvilWalletRpc` and `EthAPI` but it's a lot of annoying extra sutff that needs to be written specifically on the Reth side. Also using straigth the pool is not ideal, since there is some tx fill logic that would need to be duplicated.